docs-2.0/3.ngql-guide/11.edge-statements/3.upsert-edge.md

```diff
@@ -0,0 +1,42 @@
+# UPSERT EDGE
+
+```ngql
+UPSERT EDGE <src_vid> -> <dest_vid> [@rank] OF <edge_type> SET <update_columns> [WHEN <condition>] [YIELD <columns>]
+```
+
+- `update_columns` is the properties of the edge to be updated. For example, `e1.col1 = $^.e1.col2 + 1` means to update `e1.col1` to `e1.col2+1`.
+
+    > **NOTE:**  `$^` indicates the edge to be updated.
+
+- `condition` is some constraints. Only when met, `UPSERT` is executed successfully. `condition` supports expression operations.
+- `columns` is the columns to be returned, `YIELD` returns the latest updated values.
+
+`UPSERT` is a combination of `UPDATE` and `INSERT`.  Use `UPSERT EDGE` to update properties on an edge if it exists or insert a new edge if it does not exist. The `UPDATE EDGE` statement only updates one edge type of an edge at a time.
+
+The performance of `UPSERT` is much lower than that of `INSERT`, because `UPSERT` is a read-modify-write serialization operation at the partition level.
+
+>**DON'T:** DO NOT use `UPSERT` for scenarios with highly concurrent writes.
+
+- If the edge does not exist, a new edge is created no matter whether the condition in the `WHEN` clause is met or not. The property columns not specified by the `SET` statement use the default values of the columns. If there are no default values, an error is returned.
```

Comment:
, a new edge is created no matter whether the condition in the `WHEN` clause is met or not. ==> does that mean "the condition in the `WHEN` clause is ignored and a new edge is created"?

Reply:
You can take it this way.